test/unit/specs/components/DataCollector/index.spec.js

```diff
@@ -68,6 +68,53 @@ describe("Event Command", () => {
     });
   });
 
+  it("sends event with scopes parameter when scopes is not empty", () => {
+    const xdm = { a: "b" };
+    const data = { c: "d" };
+    const options = {
+      viewStart: true,
+      type: "test",
+      xdm,
+      data,
+      documentUnloading: true,
+      scopes: ["Foo1", "Foo2"]
+    };
+
+    return eventCommand(options).then(result => {
+      expect(event.documentMayUnload).toHaveBeenCalled();
+      expect(event.setUserXdm).toHaveBeenCalledWith(xdm);
+      expect(event.setUserData).toHaveBeenCalledWith(data);
+      expect(eventManager.sendEvent).toHaveBeenCalledWith(event, {
+        isViewStart: true,
+        scopes: ["Foo1", "Foo2"]
+      });
+      expect(result).toEqual("sendEventResult");
+    });
+  });
```

Comment:
This is totally your call to make, but I'd recommend stripping these tests down to the minimum required to test the "sends event with scopes parameter when scopes is not empty" scenario. So maybe you just have this for options:

```
const options = {
  scopes: ["Foo1", "Foo2"]
};
```

and then maybe just these two assertions:

```
expect(eventManager.sendEvent).toHaveBeenCalledWith(event, {
  isViewStart: true,
  scopes: ["Foo1", "Foo2"]
});
expect(result).toEqual("sendEventResult");
```

Reply:
make sense, thank you